### PR TITLE
add testcase for rangeForTimestamp.

### PR DESCRIPTION
**Pull request by johncming**
add testcase for rangeForTimestamp.

### CODE REVIEW

File: tsdb/db_test.go [hunk 1]
```diff
@@ -2668,3 +2668,28 @@ func TestChunkWriter(t *testing.T) {
...
 	}
 }
+
+func TestRangeForTimestamp(t *testing.T) {
+	type args struct {
+		t     int64
+		width int64
+	}
+	tests := []struct {
+		name     string
```
Comment (brian-brazil) [inline tsdb/db_test.go]: You don't need this, you can use the first value of the range.

Comment (johncming) [reply tsdb/db_test.go]: fixed.

File: tsdb/db_test.go [hunk 2]
```diff
@@ -2668,3 +2668,27 @@ func TestChunkWriter(t *testing.T) {
...
+		expected int64
+	}{
+		{args{1, 5}, 5},
+		{args{5, 5}, 10},
+		{args{6, 5}, 10},
+		{args{13, 5}, 15},
+		{args{95, 5}, 100},
+	}
+	for i, tt := range tests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```
Comment (brian-brazil) [inline tsdb/db_test.go]: Usually we wouldn't use subtests here, these are all very fast tests. It usually only comes up with benchmarks.

File: tsdb/db_test.go [hunk 3]
```diff
@@ -2668,3 +2668,27 @@ func TestChunkWriter(t *testing.T) {
...
+			if got := rangeForTimestamp(tt.args.t, tt.args.width); got != tt.expected {
+				testutil.Equals(t, tt.expected, got)
```
Comment (brian-brazil) [inline tsdb/db_test.go]: testutil.Equals does the == for you

Comment (johncming) [reply tsdb/db_test.go]: sorry, fixed.

File: tsdb/db_test.go (hunk 2 above)
Comment (johncming) [reply tsdb/db_test.go]: fixed.

File: tsdb/db_test.go [hunk 4]
```diff
@@ -2668,3 +2668,24 @@ func TestChunkWriter(t *testing.T) {
...
+func TestRangeForTimestamp(t *testing.T) {
+	type args struct {
+		t     int64
+		width int64
+	}
+	tests := []struct {
+		args     args
+		expected int64
+	}{
+		{args{1, 5}, 5},
```
Comment (bwplotka) [inline tsdb/db_test.go]: Worth to add some edge cases like `-20` or `0`?  (valid timestamps)

Comment (johncming) [reply tsdb/db_test.go]: like `t=-20, width=5`?  Is there actually a negative value here?

**Comment by brian-brazil** [comment]
Could you fix the conflicts?

**Comment by johncming** [comment]
> Could you fix the conflicts?

OK. I will do it.

**Comment by brian-brazil** [comment]
The tests are failing.

**Comment by johncming** [comment]
> The tests are failing.

Updated.

**Comment by brian-brazil** [comment]
Thanks!
